clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -441,19 +446,26 @@ const char *SYCL::Linker::constructLLVMLinkCommand(
             C.getDriver().IsCLMode())
           LibPostfix = ".obj";
       }
+      StringRef NewLibPostfix = ".new.o";
+      if (HostTC->getTriple().isWindowsMSVCEnvironment() &&
+          C.getDriver().IsCLMode())
+        NewLibPostfix = ".new.obj";
       std::string FileName = this->getToolChain().getInputFilename(II);
       StringRef InputFilename = llvm::sys::path::filename(FileName);
       if (IsNVPTX || IsSYCLNativeCPU) {
         // Linking SYCL Device libs requires libclc as well as libdevice
         if ((InputFilename.find("libspirv") != InputFilename.npos ||
              InputFilename.find("libdevice") != InputFilename.npos))
           return true;
-        if (IsNVPTX)
+        if (IsNVPTX) {
           LibPostfix = ".cubin";
+          NewLibPostfix = ".new.cubin";
```

Comment:
It looks like we miss tests with ".new.cubin".

Reply:
I will remove this line. .cubin are intermediate filenames that are created in the old offloading driver flow.